### PR TITLE
Elm test and run last

### PR DESCRIPTION
**Pull request by mbernerslee**
Adding an `-et` / `--elm-test` option that will run `elm-test`
when any `*.elm` file is saved.

**Also**

Added a feature where if the last file saved is not a recognised extension, then contant_testing will re-run the last command. Helpful if you jump to some other file which indirectly effects the test you were just writing, and you want to see what effect saving this other file has on your test.

### CODE REVIEW

File: constant_testing [hunk 1]
```diff
@@ -110,17 +106,60 @@ clear_screen() {
...
+    echo "$output"
+  fi
+  if [[ $result != 0 ]]; then
+    log_failure "$failure_message"
+  else
+    log_success "$success_message"
+  fi
+}
+
+run_last_cmd() {
```
Comment (danturn) [inline constant_testing:151]: i dont think it should do this if you have a pinned file?

i.e. if you run 

constant_testing chuffy_test.exs
 
it should always run that no matter what file you save?

File: constant_testing [hunk 2]
```diff
@@ -110,17 +106,60 @@ clear_screen() {
...
+    log_info "I don't know what to do with the file you just saved, so I'll just run the last thing I did again..."
+    run_command "$last_run_cmd" "Looks like everything is ok?" "It totally wrecked..." true
+  fi
+}
+
 watch() {
   clear_screen
-  if [ $FIXED_FILE ]; then
-    if [[ "$test_path" = *.exs ]]; then
+  if [ $ELM_TEST ]; then
```
Comment (danturn) [inline constant_testing:163]: i think its definitely time to rewrite this conditional logic...

it should be something like 

is it a fixed file? 
dont care what file type etc. is  saved, run the file

is it an elixir file
 do the elixir stuff

is it an elm file
  do the elm stuff

File: README.md [hunk 3]
```diff
@@ -22,3 +22,25 @@ Run this in your elixir project (or at the root of your umbrella project) and wh
   ```
 
 4. Start writing code!
+
+
+### Usage Varients
```
Comment (hoyon) [inline README.md:27]:
```suggestion
### Usage Variants
```

File: constant_testing [hunk 4]
```diff
@@ -110,17 +106,60 @@ clear_screen() {
...
+  elif [ $FIXED_FILE ]; then
+    if [[ "$test_path" = *.exs* ]]; then
       log_info "I'll run this for you now so you can see how broken it is before you start hacking about"
       mix_test "$test_path"
     elif [[ "$test_path" = *Main.elm ]]; then
       log_info "I'll run this for you now so you can see how broken it is before you start hacking about"
       elm_make $test_path
     else
-      log_failure "I need an exs or Main.elm file to work mate..."
+      log_failure "I don't understand the arguement you gave me. Try again buddy..."
```
Comment (hoyon) [inline constant_testing:173]:
```suggestion
      log_failure "I don't understand the argument you gave me. Try again buddy..."
```

**Review by hoyon — COMMENTED** [review]
Spelling

**Review by danturn — APPROVED** [review]

**Pull request merged**